builder/relinker/configuration.py

```
@@ -0,0 +1,301 @@
+class object_c:
+    def read_dump_info(self, paths):
+        new_env = os.environ.copy()
+        new_env['LC_ALL'] = 'C'
+        dumps = list()
+        for path in paths:
+            if not os.path.isfile(path):
+                if espidf_missing_function_info:
+                    print('Warning: object file not found, skipping: %s' % path)
+                continue
+            try:
+                dump = StringIO(subprocess.check_output([espidf_objdump, '-t', path], env=new_env).decode())
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)